Clarify CNAME move documentation

sveniu opened the pull request:
*Description of changes:*
This clarifies several steps in the ~"Move a domain name, like example.com, to another distribution"~ "Move a subdomain" section after trying to follow the steps one by one.

Please see individual commit messages for rationales.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Commit:
Rewrite to avoid "move" term

There is no single "move" operation when it comes to CloudFront
distribution alternate domain names; it's rather about removing and then
adding again.

Commit:
Remove confusing text

The "that by default is pointed to by something like" text does not read
well and is rather confusing. The `d111111abcdef8` reference is not a
clear indicator of the newly added CloudFront distribution.

Commit:
Add note on dig when using alias records

The other parts of the documentation section are careful to mention both
alias and CNAME records, but the confirmation step only includes a
description useful for CNAME records. This adds a note on what to do if
using alias records.

Commit:
Clarify DNS change vs CloudFront domain routing

When updating the DNS record to point to the new distribution, this will
not actually make the new distribution serve requests for that DNS
record. This is due to CloudFront's internal domain routing: When the
request reaches CloudFront, the request's Host header is handled by the
distribution that has a matching alternate domain name in its config.

Commit:
Add note about random routing during propagation

The additional detail about the unstable behaviour persisting even after
reaching "Deployed" status is perhaps an indication of problems with the
status being updated prematurely. It was clearly observed during
testing, however.

Commit:
Add step to remove wildcard alternate name

Removing the wildcard alternate name from the new distribution is
certainly optional (hence the "may"), but it frees up the wildcard for
use in other move operations.

joshbean commented:
Hi @sveniu –

Thank you for the PR! At first glance, this looks great 👍

One question: your PR description mentions the [Move a domain name, like example.com, to another distribution](https://github.com/awsdocs/amazon-cloudfront-developer-guide/blob/master/doc_source/CNAMEs.md#alternate-domain-names-move-domain) section, but it looks like your changes are in the [Move a subdomain name, like marketing.example.com, to another distribution](https://github.com/awsdocs/amazon-cloudfront-developer-guide/blob/master/doc_source/CNAMEs.md#alternate-domain-names-move-subdomain) section. I just want to confirm that the steps you followed were for moving a _subdomain_ from one distribution to another, is that right?

After I merge this PR, I'll go through these steps myself also, to make sure there aren't any other gotchas or issues I want to fix. I can see at least one—the `dig` example in that section shows `images.example.com`, but it should show `marketing.example.com` for consistency with the rest of the section. I'll fix that, at least 😄 

Again, thanks for the contribution! I need to stage a couple of things on my side to make sure these changes make it out to the official documentation website, and then I'll merge this PR.

joshbean commented on doc_source/CNAMEs.md:
```diff
@@ -150,11 +152,17 @@ You can’t add a wildcard to a top level domain name, such as `*.com`, so if yo
 
    The line in the Answer Section shows a CNAME resource record set that routes queries for images\.example\.com to the CloudFront distribution domain name d111111abcdef8\.cloudfront\.net\. The CNAME resource record set is configured correctly if the name on the right side of `CNAME` is the domain name for your CloudFront distribution\. If that is any other value, for example, the domain name for ...
+**Note**  
+When using Route 53 alias records, it is not possible to use dig to confirm that the resource record points to the new distribution\. In this case, you can either change the type of of the resource record from alias to CNAME, or wait until the record's time to live \(TTL\) value has expired\.
```
There's a typo here: "... you can either change the type of of the resource record ..." (double "of").

sveniu replied:
Addressed with fixup commit, that will auto-squash correctly with `git rebase --autosquash <ref>`

joshbean commented on doc_source/CNAMEs.md:
```diff
@@ -150,11 +152,17 @@ You can’t add a wildcard to a top level domain name, such as `*.com`, so if yo
-1. Remove the CNAME from the existing distribution and move it to the new CloudFront distribution\. For example, move `marketing.example.com` to a new distribution that by default is pointed to by something like `d111111abcdef8.cloudfront.net`\.
+1. Remove the CNAME from the existing distribution and then add it to the new CloudFront distribution where the wildcard alternate name was added previously\.
+**Note**  
+While these changes propagate, the alternate domain name may be served by either the original or new distribution at random\. This behaviour may even persist for a few minutes after both distributions have reached status **Deployed**\.
```
We use the US spelling ("behavior", not "behaviour") in AWS documentation.

sveniu replied:
Addressed with fixup commit, that will auto-squash correctly with `git rebase --autosquash <ref>`

Also removed the "even" word there – rationale in commit message.

Commit:
fixup! Add note on dig when using alias records

Fix "of of" typo

Commit:
fixup! Add note about random routing during propagation

Fix "behaviour" -> "behavior" typo

Commit:
fixup! Add note about random routing during propagation

Remove slightly loaded "even"

There's a slight negative tone to the "may even persist" phrasing,
admittedly because I found it to be surprising given the distribution's
status having been updated to "deployed".

Again, this might be an issue worth investigating further.

sveniu commented:
Hi, @joshbean !

Yes, this is for moving a subdomain from one distribution to another. I have updated the description now, sorry about the mix-up!

sveniu commented:
Come to think of it, a GitHub merge probably doesn't understand fixup commits for auto-squashing. If this PR is approved, I can force-push the auto-squashed commits, and you're then free to either merge as is, or do a squash to a single commit (which concatenates all the commit messages, so it's good to get rid of the fixup commits first in any case), or anything you prefer.

joshbean reviewed (approved):
This looks great! Thanks again for the contribution!

joshbean commented:
Merged as-is. I like keeping all the commits in the history because it's an accurate account of what changed, and it retains the rationale behind each change. Thanks again!
